RevenueCatUI/Templates/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaseButtonComponentView.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct PurchaseButtonComponentView: View {
+
+    private let viewModel: PurchaseButtonComponentViewModel
+
+    internal init(viewModel: PurchaseButtonComponentViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        Button {
+            // WIP: Need to perform purchase logic
+        } label: {
+            // WIP: Need to add logic for intro offer
+            Text(viewModel.cta)
+                .font(viewModel.textStyle)
+                .fontWeight(viewModel.fontWeight)
+                .fixedSize(horizontal: false, vertical: true)
+                .multilineTextAlignment(viewModel.horizontalAlignment)
+                .foregroundStyle(viewModel.color)
+                .padding(viewModel.padding)
+                .background(viewModel.backgroundColor)
+                .shape(viewModel.clipShape)
+                .applyIfLet(viewModel.cornerRadiuses, apply: { view, value in
+                    view
+                    .roundedCorner(value.topLeading, corners: .topLeft)
```

Comment:
Nit: Instead of applying four view modifiers here, could we instead do

```swift
.clipShape(    // could still use .mask()
   UnevenRoundedRectangle(
      topLeadingRadius: viewModel.topLeading,
      bottomLeadingRadius: viewModel.bottomLeading,
      bottomTrailingRadius: viewModel.bottomTrailing,
      topTrailingRadius: viewModel.topTrailing
   )
)
```

The performance difference is probably negligible, but somehow it feels better than applying 4 separate view modifiers

Reply:
Ah yes! This is actually changed in #4328 😁